planning/motion_velocity_smoother/src/smoother/smoother_base.cpp

```diff
@@ -85,12 +85,25 @@ boost::optional<TrajectoryPoints> SmootherBase::applyLateralAccelerationFilter(
   }
   output->back() = input.back();  // keep the final speed.
 
-  constexpr double curvature_calc_dist = 5.0;  // [m] calc curvature with 5m away points
-  const size_t idx_dist =
-    static_cast<size_t>(std::max(static_cast<int>((curvature_calc_dist) / points_interval), 1));
+  /*  Fill the steering tire angle w.r.t. orientation
+   *
+   *  delta_orientation / delta_time = velocity * tan(steering_tire_angle) / wheelbase
+   *
+   *  calculate desired steering_tire_angle w.r.t. delta_orientation, velocity, and wheelbase
+   *
+   */
+
+  for (size_t i = 0; i < output->size() - 1; i++) {
```

Comment:
Sorry for being annoying but I recommend using `i + 1 < output->size()` instead of `i < output->size() - 1`.
`output->size()` is a `size_t` which is an unsigned type, which can cause a critical issue in the `0 - 1` case.

Reply:
No problem, sorry for that, yes you are right. I will update immediately.